CHANGELOG.md

```diff
@@ -1,3 +1,34 @@
+## 1.59.0
+
+### Command Line Interface
+
+* New `--fatal-deprecation` flag that lets you treat a deprecation warning as
+  an error. You can pass an individual deprecation ID (e.g. `slash-div`) or you
+  can pass a Dart Sass version to treat all deprecations initially emitted in
+  that version or earlier as errors.
+
+* New `--future-deprecation` flag that lets you opt into warning for use of
+  certain features that will be deprecated in the future. At the moment, the
+  only option is `--future-deprecation=import`, which will emit warnings for
+  Sass `@import` rules, which are not yet deprecated, but will be in the future.
+
+### Dart API
+
+* New `Deprecation` enum, which contains the different current and future
+  deprecations used by the new CLI flags.
+
+* The optional `deprecation` boolean parameter of the `warn` function is now
+  deprecated. Use `deprecationType` to pass the specific `Deprecation` being
+  warned for instead.
```

Comment:
I was thinking they would pass `Deprecation.userAuthored`, but do you think it would make more sense to just leave the boolean parameter un-deprecated? (since right now if they pass `deprecation: true` with no `deprecationType`, it defaults to `Deprecation.userAuthored`)

Or could it be confusing to have both `deprecation` and `deprecationType` un-deprecated, since then you could potentially call it with a deprecation type but explicitly pass `deprecation: false`? (in which case the boolean parameter would be ignored)

Would making `deprecation` nullable but un-deprecated work?

If we do leave the `deprecation` un-deprecated in `EvaluationContext.warn`, should we also do the same for `Logger.warn`? If we do that, should we just leave the `warnForDeprecation` extension for `Logger` as private? (since custom functions should usually be just using `Deprecation.userAuthored` anyway)

Reply:
What if we just have two separate functions for both `EvaluationContext` _and_ `Logger`, one for public consumption that takes a boolean `deprecation` flag and one that's private to Sass which takes a `Deprecation`?

Reply:
Done